app/search_builders/hyrax/abstract_type_relation.rb

```diff
@@ -0,0 +1,44 @@
+module Hyrax
+  class AbstractTypeRelation < ActiveFedora::Relation
+    def initialize(opts = {})
+      super(DummyModel, opts)
+    end
+
+    def allowable_types
+      raise NotImplementedException, "Implement allowable_types in a subclass"
+    end
+
+    def equivalent_class?(klass)
+      allowable_types.include?(klass)
+    end
+
+    def search_model_clause
+      clauses = allowable_types.map do |k|
+        ActiveFedora::SolrQueryBuilder.construct_query_for_rel(has_model: k.to_s)
+      end
+      clauses.size == 1 ? clauses.first : "(#{clauses.join(' OR ')})"
```

Comment:
@dbrower this isn't logic I'm changing. I've just moved it from a concrete class to an abstract class.